pulp_2to3_migration/app/pre_migration.py

```diff
@@ -403,16 +408,29 @@
     Pulp2RepoContent.objects.bulk_create(repocontent)
 
 
-async def mark_removed_resources(plan):
+async def mark_removed_resources(plan, type_to_repo_ids):
     """
     Marks repositories, importers and distributors which are no longer present in Pulp2.
 
     Args:
         plan(MigrationPlan): A Migration Plan
+        type_to_repo_ids(dict): A mapping from a pulp 2 repo type to a list of pulp 2 repo_ids
     """
-    # Mark repositories
-    mongo_repo_types = [f'{type}-repo' for type in plan.get_plugins()]
-    mongo_repo_q = mongo_Q(__raw__={"notes._repo-type": {'$in': mongo_repo_types}})
+    repos = plan.get_repositories()
+
+    # filter by repo type
+    mongo_repo_q = None
+    repos_to_consider = []
+
+    for repo_type in plan.get_plugins():
+        repos_to_consider += type_to_repo_ids[repo_type]
+
+        # in case only certain repositories are specified in the migration plan
+        if repos:
+            repos_to_consider = set(repos).difference(repos_to_consider)
+
+        mongo_repo_q = mongo_Q(repo_id__in=repos_to_consider)
```

Comment:
This part is basically the same as in `pre_migrate_all_without_content`, maybe worth putting in a separate function.